File: src/main/java/me/sangbo/workfair/Rev1.java
package me.sangbo.workfair;

import com.rabbitmq.client.*;
import me.sangbo.ConnectionUtils;

import java.io.IOException;
import java.util.concurrent.TimeoutException;

/**
 * Created by 桑博 on 2018/6/13.
 */
public class Rev1 {


  public static final String QUEUE_NAME= "queue_work_sangbo";
  public static void main(String[] args) throws IOException, TimeoutException {

    Connection connection = ConnectionUtils.getConnection();
    Channel channel = connection.createChannel();

    channel.queueDeclare(QUEUE_NAME,false,false,false,null);

    channel.basicQos(1);

    DefaultConsumer defaultConsumer = new DefaultConsumer(channel){
      @Override
      public void handleDelivery(String consumerTag, Envelope envelope, AMQP.BasicProperties properties, byte[] body) throws IOException {

        System.out.println("rev1收到的消息："+new String(body,"utf-8"));
        try {
          Thread.sleep(1000);
        } catch (InterruptedException e) {
          e.printStackTrace();
        }finally {
          channel.basicAck(envelope.getDeliveryTag(),false);
        }
      }
    };

    boolean autoAck = false;
    channel.basicConsume(QUEUE_NAME,autoAck,defaultConsumer);



  }

}

File: src/main/java/me/sangbo/topic/Rev1.kt
package me.sangbo.topic

import com.rabbitmq.client.AMQP
import com.rabbitmq.client.DefaultConsumer
import com.rabbitmq.client.Envelope
import me.sangbo.ConnectionUtils

/**
 * Created by 桑博 on 2018/6/14.
 */

private const val EXCHANGE_NAME = "exchange_sangbo_topic"
private const val ROULTING_KEY = "goods.#";
private const val QUEUE_NAME = "queue_topic_sangbo_1"
fun main(args: Array<String>) {


    val connection = ConnectionUtils.getConnection()
    val channel = connection.createChannel()
    channel.queueDeclare(QUEUE_NAME,false,false,false,null);
    channel.exchangeDeclare(EXCHANGE_NAME,"topic")
    channel.queueBind(QUEUE_NAME, EXCHANGE_NAME, ROULTING_KEY)

    val consumer = object : DefaultConsumer(channel) {

        override fun handleDelivery(consumerTag: String?, envelope: Envelope?, properties: AMQP.BasicProperties?, body: ByteArray?) {

            println("Rev1："+String(body!!))
            Thread.sleep(1000)
            channel.basicAck(envelope!!.deliveryTag,false);

        }

    }
    val autoAck = false;
    channel.basicConsume(QUEUE_NAME,autoAck,consumer)


}